solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,406 @@
+  type        = string
+  default     = "private"
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid visibility option. Allowed values are 'public', 'private', or 'public-and-private'."
+  }
+}
+
+variable "region" {
+  description = "The IBM Cloud region where Cloud logs instance will be created."
+  type        = string
+  default     = "us-south"
+}
+
+########################################################################################################################
+# Cloud Logs
+########################################################################################################################
+
+variable "existing_cloud_logs_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of an existing Cloud Logs instance. If not supplied, a new instance will be created."
+}
+
+variable "cloud_logs_instance_name" {
+  type        = string
+  description = "The name of the IBM Cloud Logs instance to create. Defaults to 'cloud-logs-<region>'"
+  default     = null
+}
+
+variable "cloud_logs_resource_tags" {
+  type        = list(string)
+  description = "Tags associated with the IBM Cloud Logs instance (Optional, array of strings)."
+  default     = []
+}
+
+variable "cloud_logs_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the IBM Cloud Logs instance created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
+}
+
+variable "cloud_logs_retention_period" {
+  type        = number
+  description = "The number of days IBM Cloud Logs will retain the logs data in Priority insights. Allowed values: 7, 14, 30, 60, 90."
+  default     = 7
+
+  validation {
+    condition     = contains([7, 14, 30, 60, 90], var.cloud_logs_retention_period)
+    error_message = "Valid values 'cloud_logs_retention_period' are: 7, 14, 30, 60, 90"
+  }
+}
+
+########################################################################################################################
+# COS
+########################################################################################################################
+
+variable "existing_cos_instance_crn" {
+  type        = string
+  description = "The CRN of an existing Object Storage instance."
+
+  validation {
+    condition     = var.existing_cos_instance_crn == null ? var.existing_cloud_logs_crn != null : true
+    error_message = "A value must be passed for 'existing_cos_instance_crn' when creating a new instance."
+  }
+}
+
+variable "logs_cos_bucket_name" {
+  type        = string
+  nullable    = true
+  default     = "cloud-logs-logs-bucket"
+  description = "The name of an to be given to a new bucket inside the existing Object Storage instance to use for Cloud Logs."
+}
+
+variable "metrics_cos_bucket_name" {
```

Comment:
please be consistent with your variable naming. This should start with `cloud_logs_`. Perhaps `cloud_logs_metrics_cos_bucket_name`